pkg/agent/openflow/pipeline.go

```diff
@@ -41,39 +41,75 @@ import (
-	ServiceConntrackTable        = binding.NewOFTable(24, "ServiceConntrack") // serviceConntrackTable use a new ct_zone to transform SNAT'd connections.
-	ConntrackTable               = binding.NewOFTable(30, "ConntrackZone")
-	ConntrackStateTable          = binding.NewOFTable(31, "ConntrackState")
-	ServiceClassifierTable       = binding.NewOFTable(35, "ServiceClassifier")
-	SessionAffinityTable         = binding.NewOFTable(40, "SessionAffinity")
-	DNATTable                    = binding.NewOFTable(40, "DNAT(SessionAffinity)")
-	ServiceLBTable               = binding.NewOFTable(41, "ServiceLB")
-	EndpointDNATTable            = binding.NewOFTable(42, "EndpointDNAT")
-	AntreaPolicyEgressRuleTable  = binding.NewOFTable(45, "AntreaPolicyEgressRule")
-	DefaultTierEgressRuleTable   = binding.NewOFTable(49, "DefaultTierEgressRule")
-	EgressRuleTable              = binding.NewOFTable(50, "EgressRule")
-	EgressDefaultTable           = binding.NewOFTable(60, "EgressDefaultRule")
-	EgressMetricTable            = binding.NewOFTable(61, "EgressMetric")
-	L3ForwardingTable            = binding.NewOFTable(70, "L3Forwarding")
-	SNATTable                    = binding.NewOFTable(71, "SNAT")
-	L3DecTTLTable                = binding.NewOFTable(72, "IPTTLDec")
-	L2ForwardingCalcTable        = binding.NewOFTable(80, "L2Forwarding")
-	AntreaPolicyIngressRuleTable = binding.NewOFTable(85, "AntreaPolicyIngressRule")
-	DefaultTierIngressRuleTable  = binding.NewOFTable(89, "DefaultTierIngressRule")
-	IngressRuleTable             = binding.NewOFTable(90, "IngressRule")
-	IngressDefaultTable          = binding.NewOFTable(100, "IngressDefaultRule")
-	IngressMetricTable           = binding.NewOFTable(101, "IngressMetric")
-	ConntrackCommitTable         = binding.NewOFTable(105, "ConntrackCommit")
-	ServiceConntrackCommitTable  = binding.NewOFTable(106, "ServiceConntrackCommit")
-	HairpinSNATTable             = binding.NewOFTable(108, "HairpinSNAT")
-	L2ForwardingOutTable         = binding.NewOFTable(110, "Output")
+	// PipelineClassifierTable is table 0. Packets are forwarded to different pipelines in this table.
+	PipelineClassifierTable = newFeatureTable("PipelineClassifier", 0xff, 0)
+
+	// Pipeline of ARP is used to process ARP packets.
+	// ValidationStage
+	ARPSpoofGuardTable = newFeatureTable("ARPSpoofGuard", 0x7f, binding.ValidationStage)
+
+	// OutputStage
+	ARPResponderTable = newFeatureTable("ARPResponder", 0x7f, binding.OutputStage)
+
+	// Pipeline of IP is used to process IPv4/IPv6 packets.
+	// ClassifierStage
+	ClassifierTable = newFeatureTable("Classification", 0x7f, binding.ClassifierStage)
+	UplinkTable     = newFeatureTable("Uplink", 0x6f, binding.ClassifierStage)
+
+	// ValidationStage
+	SpoofGuardTable          = newFeatureTable("SpoofGuard", 0x7f, binding.ValidationStage)
+	IPv6Table                = newFeatureTable("IPv6", 0x6f, binding.ValidationStage)
+	MulticastClassifierTable = newFeatureTable("MulticastClassifier", 0x5f, binding.ValidationStage)
```

Comment:
It is a bit strange of the name "MulticastClassifierTable". I know that we currently install one flow in this table to enforce the multicast traffic to go into multicast pipeline. I am thinking if we have similar requirements for IP packets to redirect to a new pipeline with subtypes in future, we could reuse this table. So it should be better to have a more generic table name to dispatch the IP packet.

Reply:
How about `IPClassifierTable`?